### PR TITLE
Add header examples

### PR DESCRIPTION
**Pull request by JeffGreiner-eaton**
Fixes BLUI-5794.

#### Changes proposed in this Pull Request:

- add header examples "header with image needs fixed"
-
-

#### Screenshots / Screen Recording (if applicable)

image

#### To Test:

-

#### Any specific feedback you are looking for?

- need suggestion on background for header
// import BackgroundImage from '../../../assets/farm.jpg';

export const HeaderWithBackgroundExample = (): JSX.Element => (
  <ExampleShowcase>
    <Header
      title={'Valley Forge'}
      subtitle={'The Last Stand'}
      // backgroundImage={BackgroundImage} This loads image but loose styles
      // backgroundImage={require('../../../assets/farm.jpg')} This just crashes the page
    />
  </ExampleShowcase>

### CODE REVIEW

File: docs/src/componentDocs/Header/markdown/HeaderExamples.mdx [hunk 1]
```diff
@@ -1,180 +1,35 @@
...
+import {Header} from '../examples/Header';
+import {HeaderWithBackground} from '../examples/HeaderWithBackground';
+import {HeaderWithActions} from '../examples/HeaderWithActions';
+import {HeaderWithSearch} from '../examples/HeaderWithSearch';
+import {HeaderWithExpandCollapse} from '../examples/HeaderWithExpandCollapse';
 
 ## Header
 
-A basic `Header` component is used at the top of the screen and can show a title and (optional) subtitle. A third line of text can also be configured using the `info` prop, but it will only be visible when the `Header` is expanded (see below). 
+A basic `Header` component is used at the top of the screen and can show a title and (optional) subtitle. A third line of text can also be configured using the `info` prop, but it will only be visible when the `Header` is expanded (see below).
```
Comment (surajeaton) [inline docs/src/componentDocs/Header/markdown/HeaderExamples.mdx:9]: Following the conventions, the component name should be written inside angle braces `<Header>`

Comment (surajeaton) [reply docs/src/componentDocs/Header/markdown/HeaderExamples.mdx:9]: I missed this in earlier PR.

File: docs/src/componentDocs/Header/markdown/HeaderExamples.mdx [hunk 2]
```diff
@@ -1,180 +1,35 @@
...
-</Box>
-
-```
-<Header
-    title={'Valley Forge'}
-    subtitle={'The Last Stand'}
-    backgroundImage={farm.jpg}
-/>;
-```
+<HeaderWithBackground />
```
Comment (surajeaton) [inline docs/src/componentDocs/Header/markdown/HeaderExamples.mdx:17]: The code snippet has commented prop `backgroundImage`

Screenshot 2024-06-17 at 1 07 21 PM

**Comment by JeffGreiner-eaton** [comment]
Results using `backgroundImage={BackgroundImage}` loads image but we lose styling and if used like this `backgroundImage={require('../../../assets/farm.jpg')}` the image does not load

image

**Comment by joebochill** [comment]
@JeffGreiner-eaton @shubham-eaton 

This will fix the background image overflow issue:
image

The HeaderBackgroundImage component sets the background image width to the `deviceWidth`, which in dev docs is too wide because we're only showing it in the small window on the screen. Overriding this width property will make it fill properly.

**Review by surajeaton — APPROVED** [review]
Looks good

**Pull request merged**